src/include/type/types.h

```diff
@@ -728,70 +728,73 @@ std::ostream &operator<<(std::ostream &os, const ResultType &type);
 // Constraint Types
 //===--------------------------------------------------------------------===//
 
-enum PostgresConstraintType {
-  POSTGRES_CONSTRAINT_NULL, /* not standard SQL, but a lot of people * expect it
-                               */
-  POSTGRES_CONSTRAINT_NOTNULL,
-  POSTGRES_CONSTRAINT_DEFAULT,
-  POSTGRES_CONSTRAINT_CHECK,
-  POSTGRES_CONSTRAINT_PRIMARY,
-  POSTGRES_CONSTRAINT_UNIQUE,
-  POSTGRES_CONSTRAINT_EXCLUSION,
-  POSTGRES_CONSTRAINT_FOREIGN,
-  POSTGRES_CONSTRAINT_ATTR_DEFERRABLE, /* attributes for previous constraint
-                                          node */
-  POSTGRES_CONSTRAINT_ATTR_NOT_DEFERRABLE,
-  POSTGRES_CONSTRAINT_ATTR_DEFERRED,
-  POSTGRES_CONSTRAINT_ATTR_IMMEDIATE
+enum class PostgresConstraintType {
+  NULL_TYPE, /* not standard SQL, but a lot of people * expect it */
+  NOTNULL,
+  DEFAULT,
+  CHECK,
+  PRIMARY,
+  UNIQUE,
+  EXCLUSION,
+  FOREIGN,
+  ATTR_DEFERRABLE, /* attributes for previous constraint node */
+  ATTR_NOT_DEFERRABLE,
+  ATTR_DEFERRED,
+  ATTR_IMMEDIATE
 };
 
-enum ConstraintType {
-  CONSTRAINT_TYPE_INVALID = INVALID_TYPE_ID,  // invalid
-  CONSTRAINT_TYPE_NULL = 1,                   // notnull
-  CONSTRAINT_TYPE_NOTNULL = 2,                // notnull
-  CONSTRAINT_TYPE_DEFAULT = 3,                // default
-  CONSTRAINT_TYPE_CHECK = 4,                  // check
-  CONSTRAINT_TYPE_PRIMARY = 5,                // primary key
-  CONSTRAINT_TYPE_UNIQUE = 6,                 // unique
-  CONSTRAINT_TYPE_FOREIGN = 7,                // foreign key
-  CONSTRAINT_TYPE_EXCLUSION = 8               // foreign key
+enum class ConstraintType {
+  INVALID = INVALID_TYPE_ID,  // invalid
+  NULL_TYPE = 1,              // notnull
```

Comment:
@apavlo I am not sure what exactly is `CONSTRAINT_TYPE_NULL` here and why it also represents notnull. Since NULL is a keyword itself, NULL here should be renamed and I am wondering what name best represents this type?